### ISSUE TITLE
Feedback: Search - missing visible label

### ISSUE THREAD

**Issue by sigveres** [digdir/designsystemet#4515]
The component only uses `aria-label` to provide an accessible name, but does not include a visible `<label>` element. It may create usabiliy challanges for users who rely on speech recognition or cognitive support. Placeholder text is not considered a sufficient substitute for label, it can also introduce contrast issues depending on styling. This is related to ([WCAG 3.3.2](https://www.uutilsynet.no/wcag-standarden/332-ledetekster-eller-instruksjoner-niva/117)). We should also be aware for [WCAG 2.5.3](https://www.uutilsynet.no/wcag-standarden/253-ledetekst-i-navn-niva/150) if an visible label is present to ensure compatibility with speech recognition. 

My rule of thumb: Every input should have a visible and unique label

**Comment by Barsnes** [comment]
Hi!
Consumers can freely add a label to the input:

<img width="1000" height="639" alt="Image" src="https://github.com/user-attachments/assets/1ecd40b9-e20b-40a5-b8ea-8412c0e9e340" />

I think it's good for us to leave this issue though, since it might be relevant to talk about this on the a11y page

**Comment by Febakke** [comment]
I agree that `Placeholder` does not replace label. But a Label do not always have to visually be a "form label". Meaning placed on the left side or over the inputfield. 

WCAGs definition of Label is broader

> Text or other component with a text alternative that is presented to a user to identify a component within web content

[Source](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions.html#dfn-label)

In the case of searchfields, the most common pattern in my experience is to use a button or "magnifying-glass" icon to indicate that this is a search field. (I think context also is a big part of this when it comes to search) Using other elements than traditional text is acceptable when talking about 3.3.2 Labels and instructions. 

So my reading of 3.3.2 would look something like this: 
<img width="450" height="236" alt="Illustration showing how icon or button is acceptable designs, but only placeholder text would fail 3.3.2 Labels and instructions" src="https://github.com/user-attachments/assets/6939625d-bb3b-4461-9750-3c6530d0fe4e" />
**Note:** Might be depending on contrast on placeholder if you can use it as visible label? Im not sure, but either way it will vanish when it gets focus so I guess that rules it out 


That said, I cant think of any other situation where I would use icon to represent a label for input fields. Maybe in "expert" ui? 

2.5.3 Label in Name is interesting. Im guessing that having the accessible label be "search" is the common pattern here? We should do some research on this. 

We will document this when working on #4286
